gobblin-utility/src/main/java/org/apache/gobblin/util/WriterUtils.java

```diff
@@ -92,6 +92,17 @@ public static Path getWriterStagingDir(State state, int numBranches, int branchI
         WriterUtils.getWriterFilePath(state, numBranches, branchId));
   }
 
+  public static Path getDatasetWriterStagingDir(State state, int numBranches, int branchId) {
```

Comment:
You would like to reduce the code duplication here and re-use the method that is already implemented above.